src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -215,14 +215,20 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
             return null;
         }
 
-        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out IntPtr dstIntPtr)
+        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out object dstIntPtr)
         {
             if (srcObject is IntPtr srcIntPtr)
             {
                 dstIntPtr = srcIntPtr;
```

Comment:
```suggestion
            if (srcObject is IntPtr)
            {
                dstPtr = srcObject;
```
You can just reuse the incoming box without reboxing it again